### DISCUSSION TITLE
Task 2, Lab 03.02

### DISCUSSION THREAD

**Discussion by ThomasPineda** [rramosp/ai4eng.v1#200] [03 - GENERACION DE MODELOS]
mi función es la siguiente:

def apply_linear_regression_model(td, w):
  import numpy as np
  lista_nom = []
  s = np.array(0)

  for i in range(len(td.columns) - 2):
    nombre = "signal -"+str(i+1)
    lista_nom.append(nombre)

  for i in range(len(w[2:])):
    s = s + w[i]*td[lista_nom[i]]
   s = s + w[0] + td['signal']*w[1]
  

  r = np.array(s) # YOUR SOLUTION HERE
  return r
![image](https://user-images.githubusercontent.com/86691796/190013295-dd07a2ba-800b-4b74-92fd-6ee027f9427d.png)

cuando pruebo el codígo me funciona
![image](https://user-images.githubusercontent.com/86691796/190012627-60f99292-7fc7-4c55-8452-14edb704fbc5.png)

pero me sale este error al revisar la tarea y no entiendo por qué
![image](https://user-images.githubusercontent.com/86691796/190012739-1eb946a3-d381-405f-8843-f6f3affec608.png)
![image](https://user-images.githubusercontent.com/86691796/190012782-e490f963-9a45-4af3-8650-ddf67544e394.png)

**Comment by Felipe-RA** [comment]
Saludos @ThomasPineda 

El error que tienes es un error de tipo KeyError. Significa que el grader intenta acceder un diccionario con una clave que no existe.

Esto ocurre porque estas nombrando las señales de una forma incorrecta, revisa tus signal

También, por favor usa el formato de bloque de código para facilitar la lectura de tu código. Lo puedes hacer usando las comillas tal cual como te lo pongo más adelante (sin los #)

```

# ```
# ```
```
